pql/parser.go

```diff
@@ -72,7 +72,8 @@ func (p *parser) Parse() (*Query, error) {
 		p.Execute()
 	}()
 	if v != nil {
-		if strings.HasPrefix(v.(string), duplicateArgErrorMessage) || strings.HasPrefix(v.(string), outOfBoundsErrorMessage) {
+		errorMessage := v.(string)
```

Comment:
ah, what we can do here to be a little bit more robust to this kind of problem in the future is:

```
errorMessage, ok := v.(string)
if !ok {
    return nil, fmt.Errorf("unexpected parser error of type %T: %[1]v", v)
}
```